public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -448,6 +448,7 @@ export class PanelModel implements DataConfigSource {
 
   setTransformations(transformations: DataTransformerConfig[]) {
     this.transformations = transformations;
+    this.resendLastResult();
```

Comment:
I get why we need this now, but moving forward shouldn't we subscribe to changes instead of pulling changes?

Reply:
That sounds good to me. But I am new to this part of the code so not sure what the plan is.

Reply:
@hugohaggmark the panel editor & panel are subscribing to data changes (via rxjs), this call is to tell the PanelQueryRunner to send the last result on the subject again so that the panel get's updated data and re-renders

Reply:
so I am not sure we should subscribe to here, do you think transformations array should be an Observable?

Reply:
yes, but I don't have the complete picture at all. This just looked like a bad pattern.

Reply:
yea agree, but not sure I see a way around it without using some kind of observable patterns for state